docs/dev-practices.md

```diff
@@ -116,3 +116,15 @@ be sure to have [enabled custom
 formatters](https://github.com/binaryage/cljs-devtools/blob/master/docs/installation.md#enable-custom-formatters-in-chrome)
 in the desktop app and browser. Without this enabled, most of the log messages
 aren't readable.
+
+## Data validation and generation
+
+We currently use [spec](https://github.com/clojure/spec.alpha) for data
+validation (and generation someday). We may switch to
+[malli](https://github.com/metosin/malli) if we need to datafy our data models
+at some point.
+
+Specs should go under `src/main/frontend/spec/` and be compatible with clojure
```

Comment:
There is more I'd like to write here but I think we need to encounter other use cases I'm thinking of in order to understand the recommended practices. Hopefully the PR demonstrates why I make this recommendation